src/plugins/cams/datasource.py

```
@@ -0,0 +1,509 @@
+            raise DataSourceError("No CAMS data available for the requested period")
+            
+        # Combine datasets
+        combined = xr.concat(datasets, dim='time')
+        combined = combined.sortby('time')
+        
+        # Cache the combined dataset
+        self._dataset_cache[cache_key] = combined
+        
+        return combined
+        
+    async def get_measurements(
```

Ruff (0.12.2)

`get_measurements` is too complex (11 > 10)

(C901)